sdk/inc/azure/core/az_result.h

```diff
@@ -133,26 +144,32 @@ typedef enum
       8), ///< Generic error in the HTTP transport adapter implementation.
 
   // IoT error codes
+
+  /// The IoT topic is not matching the expected format.
   AZ_ERROR_IOT_TOPIC_NO_MATCH = _az_RESULT_MAKE_ERROR(_az_FACILITY_IOT, 1),
 } az_result;
 
-/// Checks whether the \a result provided indicates a failure.
-///
-/// @param result Result value to check for failure.
-///
-/// @retval true \a result indicates a failure.
-/// @retval false \a result is successful.
+/**
+ * @brief Checks whether the \a result provided indicates a failure.
+ *
+ * @param[in] result Result value to check for failure.
+ *
+ * @retval true \a result indicates a failure.
+ * @retval false \a result is successful.
+ */
 AZ_NODISCARD AZ_INLINE bool az_failed(az_result result)
```

Comment:
@JeffreyRichter - do we need both `az_failed` and `az_succeeded` or can we just have `az_succeeded`?

Semantically, isn't one just a negation of the other. Why have both?